mvg/analysis_classes.py

```diff
@@ -100,21 +133,30 @@ def _add_datetime_df(self, dframe, timecolumn):
         """
 
         # Check if there is info for conversion
-        if self.t_zone is not None:
+        if self._t_zone is not None:
             # EPOCH to datetime
             dframe = dframe.assign(
-                datetime=pd.to_datetime(dframe[timecolumn], unit=self.t_unit, utc=True)
+                datetime=pd.to_datetime(dframe[timecolumn], unit=self._t_unit, utc=True)
             )
             # Set timezone
             dframe = dframe.assign(
-                datetime=(dframe["datetime"].dt.tz_convert(self.t_zone))
+                datetime=(dframe["datetime"].dt.tz_convert(self._t_zone))
             )
             # Mark datetime as availble
             self.time_column = "datetime"
 
         return dframe
 
     # Accessor functions
+    def raw_results(self):
+        """Raw results as returned by server
+        Returns
+        -------
+        raw_results: dict
+        """
+
+        return self._raw_results
+
```

Comment:
I don't really see the point in making accessor functions instead of just letting them be attributes. If they have to be private for some reason I would prefer if they were properties rather than functions
```suggestion
    @property
    def raw_results(self):
        """Raw results as returned by server
        Returns
        -------
        raw_results: dict
        """

        return self._raw_results

```

Then the data can be accessed with `analysis.raw_results`, but you cannot assign values to it without an exception.

Reply:
### ABC
ABC classes shall contain no implementation, why Analysis contains some implementation of methods which relate to the general structure of the returned results. I think ABC classes are mostly used for specifying what in Java in a Interface, which is not what Analysis class is.

### accessors vs. attributes
In OO Design, getter functions are preferred to enforce encapsulation and hiding of internal structure, so that is on favour of accessor functions. If I move over to @property, I basically have to revert a lot of changes (functions to properties) for syntactic sugar (results['key'] instead of results()[]) and update all TC and examples for analysis classes. Also I think it is good to be clear that something is a function (under the hood) rather than a property.
Next time I may go directly for @property.